libnetwork/driver_plugin.py

```diff
@@ -306,3 +531,63 @@ def remove_veth(name):
         netns.remove_veth(name)
     except CalledProcessError:
         app.logger.warn("Failed to delete veth %s", name)
+
+
+def get_network_data(network_id):
+    """
+    Return the network data, and raise an exception if the network does not
+    exists.
+
+    :param network_id: The network ID.
+    :return: The network data.
+    """
+    network_data = client.get_network(network_id)
+    if not network_data:
+        error_message = "Network %s does not exist" % network_id
+        app.logger.error(error_message)
+        raise Exception(error_message)
+    return network_data
+
+
+def get_gateway_pool_from_network_data(network_data, version):
+    """
+    Extract the gateway and pool from the network data.
+    :param network_data: The network data.
+    :param version: The IP version (4 or 6)
+    :return: Tuple of (string gateway_CIDR,
+                       string pool_CIDR)
+             or (None, None) if either are not set.
+    """
+    assert version in [4,6]
+
+    ip_data = network_data.get("IPv%sData" % version)
+    if not ip_data:
+        # No IP data for this IP version, so skip.
+        app.logger.info("No IPv%s data", version)
+        return None, None
+
+    if len(ip_data) > 1:
+        error_message = "Unsupported: mutliple Gateways defined for " \
+                        "IPv%s" % version
+        app.logger.error(error_message)
+        raise Exception(error_message)
+
+    gateway = ip_data[0].get('Gateway')
+    pool = ip_data[0].get('Pool')
+
+    if not gateway or not pool:
+        return None, None
+
+    return IPNetwork(gateway), IPNetwork(pool)
+
+
+def is_using_calico_ipam(gateway_cidr):
+    """
+    Determine if the gateway CIDR indicates that we are using Calico IPAM
+    driver for IP assignment.
+    :param gateway: The gateway CIDR.  This should not be None.
+    :return: True, if using Calico IPAM driver.
+    """
+    # A 0 gateway IP indicates Calico IPAM is being used.
+    assert isinstance(gateway_cidr, IPNetwork)
+    return (not gateway_cidr.ip) and not (gateway_cidr.prefixlen)
```

Comment:
What's the reasoning behind doing this boolean comparison? Can't you just do an equality check on an IPNetwork object directly?